Rename dataset related python variable names to asset

Lee-W opened the pull request:
## Why
as part of AIP-74

part of https://github.com/apache/airflow/issues/42307

## What

Rename the variable names in Airflow. DB, API, and UI changes will be done in the following PRs

* Rename module ``airflow.api_connexion.schemas.dataset_schema`` as ``airflow.api_connexion.schemas.asset_schema``

  * Rename variable ``create_dataset_event_schema`` as ``create_asset_event_schema``
  * Rename variable ``dataset_collection_schema`` as ``asset_collection_schema``
  * Rename variable ``dataset_event_collection_schema`` as ``asset_event_collection_schema``
  * Rename variable ``dataset_event_schema`` as ``asset_event_schema``
  * Rename variable ``dataset_schema`` as ``asset_schema``
  * Rename class ``TaskOutletDatasetReferenceSchema`` as ``TaskOutletAssetReferenceSchema``
  * Rename class ``DagScheduleDatasetReferenceSchema`` as ``DagScheduleAssetReferenceSchema``
  * Rename class ``DatasetAliasSchema`` as ``AssetAliasSchema``
  * Rename class ``DatasetSchema`` as ``AssetSchema``
  * Rename class ``DatasetCollection`` as ``AssetCollection``
  * Rename class ``DatasetEventSchema`` as ``AssetEventSchema``
  * Rename class ``DatasetEventCollection`` as ``AssetEventCollection``
  * Rename class ``DatasetEventCollectionSchema`` as ``AssetEventCollectionSchema``
  * Rename class ``CreateDatasetEventSchema`` as ``CreateAssetEventSchema``

* Rename module ``airflow.datasets`` as ``airflow.assets``

  * Rename class ``DatasetAlias`` as ``AssetAlias``
  * Rename class ``DatasetAll`` as ``AssetAll``
  * Rename class ``DatasetAny`` as ``AssetAny``
  * Rename function ``expand_alias_to_datasets`` as ``expand_alias_to_assets``
  * Rename class ``DatasetAliasEvent`` as ``AssetAliasEvent``

    * Rename method ``dest_dataset_uri`` as ``dest_asset_uri``

  * Rename class ``BaseDataset`` as ``BaseAsset``

    * Rename method ``iter_datasets`` as ``iter_assets``
    * Rename method ``iter_dataset_aliases`` as ``iter_asset_aliases``

  * Rename class ``Dataset`` as ``Asset``

    * Rename method ``iter_datasets`` as ``iter_assets``
    * Rename method ``iter_dataset_aliases`` as ``iter_asset_aliases``

  * Rename class ``_DatasetBooleanCondition`` as ``_AssetBooleanCondition``

    * Rename method ``iter_datasets`` as ``iter_assets``
    * Rename method ``iter_dataset_aliases`` as ``iter_asset_aliases``

* Rename module ``airflow.datasets.manager`` as ``airflow.assets.manager``

  * Rename variable ``dataset_manager`` as ``asset_manager``
  * Rename function ``resolve_dataset_manager`` as ``resolve_asset_manager``
  * Rename class ``DatasetManager`` as ``AssetManager``

      * Rename method ``register_dataset_change`` as ``register_asset_change``
      * Rename method ``create_datasets`` as ``create_assets``
      * Rename method ``register_dataset_change`` as ``notify_asset_created``
      * Rename method ``notify_dataset_changed`` as ``notify_asset_changed``

* Rename module ``airflow.models.dataset`` as ``airflow.models.asset``

    * Rename class ``DatasetDagRunQueue`` as ``AssetDagRunQueue``
    * Rename class ``DatasetEvent`` as ``AssetEvent``
    * Rename class ``DatasetModel`` as ``AssetModel``
    * Rename class ``DatasetAliasModel`` as ``AssetAliasModel``
    * Rename class ``DagScheduleDatasetReference`` as ``DagScheduleAssetReference``
    * Rename class ``TaskOutletDatasetReference`` as ``TaskOutletAssetReference``
    * Rename class ``DagScheduleDatasetAliasReference`` as ``DagScheduleAssetAliasReference``

* Rename module ``airflow.api_ui.views.datasets`` as ``airflow.api_ui.views.assets``

    * Rename variable ``dataset_router`` as ``asset_rounter``

* Rename module ``airflow.listeners.spec.dataset`` as ``airflow.listeners.spec.asset``

    * Rename function ``on_dataset_created`` as ``on_asset_created``
    * Rename function ``on_dataset_changed`` as ``on_asset_changed``

* Rename module ``airflow.timetables.datasets`` as ``airflow.timetables.assets``

    * Rename class ``DatasetOrTimeSchedule`` as ``AssetOrTimeSchedule``

* Rename module ``airflow.serialization.pydantic.dataset`` as ``airflow.serialization.pydantic.asset``

    * Rename class ``DagScheduleDatasetReferencePydantic`` as ``DagScheduleAssetReferencePydantic``
    * Rename class ``TaskOutletDatasetReferencePydantic`` as ``TaskOutletAssetReferencePydantic``
    * Rename class ``DatasetPydantic`` as ``AssetPydantic``
    * Rename class ``DatasetEventPydantic`` as ``AssetEventPydantic``

* Rename module ``airflow.datasets.metadata`` as ``airflow.assets.metadata``

* In module ``airflow.jobs.scheduler_job_runner``

    * and its class ``SchedulerJobRunner``

        * Rename method ``_create_dag_runs_dataset_triggered`` as ``_create_dag_runs_asset_triggered``
        * Rename method ``_orphan_unreferenced_datasets`` as ``_orphan_unreferenced_datasets``

* In module ``airflow.api_connexion.security``

    * Rename decorator ``requires_access_dataset`` as ``requires_access_asset``

* In module ``airflow.auth.managers.models.resource_details``

    * Rename class ``DatasetDetails`` as ``AssetDetails``

* In module ``airflow.auth.managers.base_auth_manager``

    * Rename function ``is_authorized_dataset`` as ``is_authorized_asset``

* In module ``airflow.timetables.simple``

    * Rename class ``DatasetTriggeredTimetable`` as ``AssetTriggeredTimetable``

* In module ``airflow.lineage.hook``

    * Rename class ``DatasetLineageInfo`` as ``AssetLineageInfo``

        * Rename attribute ``dataset`` as ``asset``

    * In its class ``HookLineageCollector``

        * Rename method ``create_dataset`` as ``create_asset``
        * Rename method ``add_input_dataset`` as ``add_input_asset``
        * Rename method ``add_output_dataset`` as ``add_output_asset``
        * Rename method ``collected_datasets`` as ``collected_assets``

* In module ``airflow.models.dag``

    * Rename function ``get_dataset_triggered_next_run_info`` as ``get_asset_triggered_next_run_info``

    * In its class ``DagModel``

        * Rename method ``get_dataset_triggered_next_run_info`` as ``get_asset_triggered_next_run_info``

* In module ``airflow.models.taskinstance``

    * and its class ``TaskInstance``

        * Rename method ``_register_dataset_changes`` as ``_register_asset_changes``

* In module ``airflow.providers_manager``

    * and its class ``ProvidersManager``

        * Rename method ``initialize_providers_dataset_uri_resources`` as ``initialize_providers_asset_uri_resources``
        * Rename attribute ``_discover_dataset_uri_resources`` as ``_discover_asset_uri_resources``
        * Rename property ``dataset_factories`` as ``asset_factories``
        * Rename property ``dataset_uri_handlers`` as ``asset_uri_handlers``
        * Rename property ``dataset_to_openlineage_converters`` as ``asset_to_openlineage_converters``

* In module ``airflow.security.permissions``

    * Rename constant ``RESOURCE_DATASET`` as ``RESOURCE_ASSET``

* In module ``airflow.serialization.enums``

    * and its class DagAttributeTypes

        * Rename attribute ``DATASET_EVENT_ACCESSORS`` as ``ASSET_EVENT_ACCESSORS``
        * Rename attribute ``DATASET_EVENT_ACCESSOR`` as ``ASSET_EVENT_ACCESSOR``
        * Rename attribute ``DATASET`` as ``ASSET``
        * Rename attribute ``DATASET_ALIAS`` as ``ASSET_ALIAS``
        * Rename attribute ``DATASET_ANY`` as ``ASSET_ANY``
        * Rename attribute ``DATASET_ALL`` as ``ASSET_ALL``

* In module ``airflow.serialization.pydantic.taskinstance``

    * and its class ``TaskInstancePydantic``

        * Rename method ``_register_dataset_changes`` as ``_register_dataset_changes``

* In module ``airflow.serialization.serialized_objects``

    * Rename function ``encode_dataset_condition`` as ``encode_asset_condition``
    * Rename function ``decode_dataset_condition`` as ``decode_asset_condition``

* In module ``airflow.timetables.base``

    * Rename class ```_NullDataset``` as ```_NullAsset```

        * Rename method ``iter_datasets`` as ``iter_assets``

* In module ``airflow.utils.context``

    * Rename class ``LazyDatasetEventSelectSequence`` as ``LazyAssetEventSelectSequence``

* In module ``airflow.www.auth``

    * Rename function ``has_access_dataset`` as ``has_access_asset``

* Rename configuration ``core.strict_dataset_uri_validation`` as ``core.strict_asset_uri_validation``, ``core.dataset_manager_class`` as ``core.asset_manager_class`` and ``core.dataset_manager_class`` as ``core.asset_manager_class``
* Rename example dags  ``example_dataset_alias.py``, ``example_dataset_alias_with_no_taskflow.py``, ``example_datasets.py`` as ``example_asset_alias.py``, ``example_asset_alias_with_no_taskflow.py``, ``example_assets.py``
* Rename DagDependency name ``dataset-alias``, ``dataset`` as ``asset-alias``, ``asset``
* Rename context key ``triggering_dataset_events`` as ``triggering_asset_events``
* Rename resource key ``dataset-uris`` as ``asset-uris`` for providers amazon, common.io, mysql, fab, postgres, trino

* In provider ``airflow.providers.amazon.aws``

    * Rename package ``datasets`` as ``assets``

        * In its module ``s3``

            * Rename method ``create_dataset`` as ``create_asset``
            * Rename method ``convert_dataset_to_openlineage`` as ``convert_asset_to_openlineage``

  * and its module ``auth_manager.avp.entities``

    * Rename attribute ``AvpEntities.DATASET`` as ``AvpEntities.ASSET``

  * and its module ``auth_manager.auth_manager.aws_auth_manager``

    * Rename function ``is_authorized_dataset`` as ``is_authorized_asset``

* In provider ``airflow.providers.common.io``

  * Rename package ``datasets``  as ``assets``

    * in its module ``file``

        * Rename method ``create_dataset`` as ``create_asset``
        * Rename method ``convert_dataset_to_openlineage`` as ``convert_asset_to_openlineage``

* In provider ``airflow.providers.fab``

  * in its module ``auth_manager.fab_auth_manager``

    * Rename function ``is_authorized_dataset`` as ``is_authorized_asset``

* In provider ``airflow.providers.openlineage``

  * in its module ``utils.utils``

    * Rename class ``DatasetInfo`` as ``AssetInfo``
    * Rename function ``translate_airflow_dataset`` as ``translate_airflow_asset``

* Rename package ``airflow.providers.postgres.datasets`` as ``airflow.providers.postgres.assets``
* Rename package ``airflow.providers.mysql.datasets`` as ``airflow.providers.mysql.assets``
* Rename package ``airflow.providers.trino.datasets`` as ``airflow.providers.trino.assets``
* Add module ``airflow.providers.common.compat.assets``
* Add module ``airflow.providers.common.compat.openlineage.utils.utils``
* Add moddule ``airflow.providers.common.compat.security.permissions.RESOURCE_ASSET``

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

Lee-W committed:
feat(assets): rename airflow.datasets as airflow.assets

uranusjr commented on airflow/providers/amazon/aws/datasets/s3.py:
We should likely make providers import from `airflow.providers.common.compat` instead. These imports need to be compatible with Airflow 2.

Lee-W replied:
Yes, it's indeed something missed. Thanks for reminding me. Will do it today

Lee-W committed:
feat: rename DatasetManager as AssetManger

Lee-W committed:
feat(assets/manager): rename create_datasets as create_assets

Lee-W committed:
feat(assets/manager): rename register_dataset_change as register_asset_change

Lee-W committed:
feat(assets/manager): rename notify_dataset_created as notify_asset_created

Lee-W committed:
feat(liserners/spec/dataset): rename on_dataset_created as on_asset_created

Lee-W committed:
feat(listeners/spec/dataset): rename on_dataset_changed as on_asset_changed

Lee-W committed:
feat(assets/manager): rename dataset variables in _queue_dagrun as asset

Lee-W committed:
feat(assets/manager): rename resolve_dataset_manager as resolve_asset_manager

also rename config dataset_manager_class and dataset_manager_kwargs as asset_manager_class and asset_manager_kwargs

Lee-W committed:
feat(assets/manager): rename exported dataset_manager as asset_manager

Lee-W committed:
feat(providers/common/compat/assets): add common.compat.assets for provider compatibility

Lee-W committed:
feat(providers/amazon): use common.compat to import Dataset

Lee-W committed:
feat(providers/common/io): use common.compat to import Dataset

Lee-W committed:
feat(providers/openlineage): use common.compat to import Dataset

Lee-W committed:
feat(assets): rename BaseDataset as BaseAsset

Lee-W committed:
feat(assets): rename _DatasetBooleanCondition as _AssetBooleanCondition

Lee-W committed:
feat(assets): rename DatasetAny as AssetAny

Lee-W committed:
feat(assets): rename DatasetAll as AssetAll

Lee-W committed:
feat(assets): rename method iter_datasets as iter_assets

Lee-W committed:
feat(assets): rename method iter_dataset_aliases as iter_asset_aliases

Lee-W committed:
fix(providers/common/compat/assets): import dataset alias only for airflow version >= 2.10

Lee-W committed:
fix(providers/common/compat/assets): import dataset any, dataset all only if airflow version >= 2.9

Lee-W committed:
feat(assets): replace part of the dataset log messages as asset in _sanitize_uri

Lee-W committed:
feat(assets): rename DatasetAliasEvent as AssetAliasEvent

Lee-W committed:
feat(models/dataset): rename DatasetAliasModel as AssetAliasModel

Lee-W committed:
feat(assets): rename _DatasetAliasCondition as _AssetAliasCondition

Lee-W committed:
feat(models/dataset): rename DagScheduleDatasetAliasReference as DagScheduleAssetAliasReference

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename DatasetAliasSchema as AssetAliasSchema

Lee-W committed:
feat(assets): rename DatasetAlias as AssetAlias

Lee-W committed:
feat(assets): rename DagScheduleDatasetReference with DagScheduleAssetReference

Lee-W committed:
feat(assets): rename TaskOutletDatasetReference as TaskOutletAssetReference

Lee-W committed:
feat(assets): rename DatasetDagRunQueue as AssetDagRunQueue

Lee-W committed:
feat(assets): rename DatasetOrTimeSchedule as AssetOrTimeSchedule

Lee-W committed:
feat(assets): rename DatasetTriggeredTimetable as AssetTriggeredTimetable

Lee-W committed:
feat(assets): rename DatasetLineageInfo as AssetLineageInfo

Lee-W committed:
feat(assets): rename DatasetModel as AssetModel

Lee-W committed:
feat(providers/amazon): rename add_input_dataset as add_input_asset

Lee-W committed:
feat(assets): rename DatasetEvent as AssetEvent

Lee-W committed:
feat(assets): rename LazyDatasetEventSelectSequence.from_select as LazyAssetEventSelectSequence.from_select

Lee-W committed:
feat(assets/manager): rename notify_dataset_changed as notify_asset_changed

Lee-W committed:
feat(assets): rename dataset.updates as asset.updates in logging metrics

Lee-W committed:
feat(assets): rename DatasetDetails as AssetDetails

Lee-W committed:
feat(providers/common/compat/assets): add AssetDetails to compat

Lee-W committed:
feat(providers/amazon): rename DatasetDetails as AssetDetails

Lee-W committed:
feat(timtables/base): rename _NullDataset as _NullAsset

Lee-W committed:
feat(providers/amazon): rename AvpEntities.DATASET as AvpEntities.ASSET

Lee-W committed:
feat(providers/fab): rename DatasetDetails as AssetDetails

Lee-W committed:
docs(airflow/example_dags): rename dataset example dags as asset example dags

Lee-W committed:
feat(models/asset): rename models.dataset as models.asset

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename DatasetEventCollectionSchema as AssetEventCollectionSchema

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename CreateDatasetEventSchema as CreateAssetEventSchema

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename DatasetEventCollection as AssetEventCollection

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename DatasetEventSchema as AssetEventSchema

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename DatasetCollectionSchema as AssetCollectionSchema

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename DatasetEventCollection as AssetEventCollection

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename DatasetSchema as AssetSchema

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename dataset_collection_schema as asset_collection_schema

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename create_dataset_event_schema as create_asset_event_schema

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename dataset_event_collection_schema as asset_event_collection_schema

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename dataset_event_schema as asset_event_schema

Lee-W committed:
feat(api_connexion/schema/dataset_schema): rename dataset_schema as asset_schema

Lee-W committed:
feat(api_connexion/schemas/asset_schema): rename module dataset_schema as asset_schema

Lee-W committed:
feat(providers/openlineage): rename collected_datasets as collected_assets

Lee-W committed:
docs(newsfragements): initialize changes

Lee-W committed:
feat(timetables): rename airflow.timetables.datasets as airflow.timetables.assets

Lee-W committed:
feat(models/dags): rename get_dataset_triggered_next_run_info as get_asset_triggered_next_run_info

Lee-W committed:
feat(airflow/assets): rename Dataset as Asset

Lee-W committed:
feat(config_templates): rename strict_dataset_uri_validation as strict_asset_uri_validation

Lee-W committed:
feat(airflow/serialization/pydantic): rename DatasetEventPydantic as AssetEventPydantic

Lee-W committed:
feat(airflow/serialization/pydantic): rename DatasetPydantic as AssetPydantic

Lee-W committed:
feat(airflow/serialization): rename DAT.DATA_SET as DAT.ASSET

Lee-W committed:
feat(dag_dependency): rename DepDependency value dataset and dataset-alias to asset and asset-alias

Lee-W committed:
feat(api_connexion/security): rename requires_access_dataset as requires_access_asset

Lee-W committed:
feat(providers_manager): rename _dataset_uri_handlers as _asset_uri_handlers

Lee-W committed:
feat(providers_manager): rename dataset_factories as asset_factories

Lee-W committed:
feat(providers_manager): rename dataset_to_openlineage_converters as asset_to_openlineage_converters

Lee-W committed:
feat(jobs/scheduler_job_runner): rename _create_dag_runs_dataset_triggered as _create_dag_runs_asset_triggered

Lee-W committed:
feat(providers_manager): rename initialize_providers_dataset_uri_resources as initialize_providers_asset_uri_resources

Lee-W committed:
feat(providers_manager): rename _discover_dataset_uri_resources as _discover_asset_uri_resources

Lee-W committed:
docs(example_dags): rename dataset in example_outlet_event_extra.py as asset

Lee-W committed:
feat(jobs/scheduler_job_runner): rename _orphan_unreferenced_datasets as _orphan_unreferenced_assets

Lee-W committed:
feat(providers/common): rename convert_dataset_to_openlineage as convert_asset_to_openlineage

Lee-W committed:
feat(providers/amazon): rename convert_dataset_to_openlineage as convert_asset_to_openlineage

Lee-W committed:
feat(providers/openlineage): rename translate_airflow_dataset as translate_airflow_asset

Lee-W committed:
feat(providers/amazon): rename create_dataset as create_asset

Lee-W committed:
feat(providers/common/io): rename create_dataset as create_asset

Lee-W committed:
feat(providers/amazon): rename airflow.providers.amazon.aws.datasets as airflow.providers.amazon.aws.assets

Lee-W committed:
feat(providers/common/io): rename airflow.providers.common.io.datasets as airflow.providers.common.io.asset

Lee-W committed:
feat(www): rename has_access_dataset as has_access_asset

Lee-W committed:
feat(security): rename RESOURCE_DATASET as RESOURCE_ASSET

Lee-W committed:
feat(providers/fab): rename RESOURCE_DATASET as RESOURCE_ASSET

Lee-W committed:
feat(providers/amazon): rename RESOURCE_DATASET as RESOURCE_ASSET

Lee-W committed:
feat(security): rename is_authorized_dataset as is_authorized_asset

Lee-W committed:
feat(providers/amazon): rename is_authorized_dataset as is_authorized_asset

Lee-W committed:
feat(providers/fab): rename is_authorized_dataset as is_authorized_asset

Lee-W committed:
feat(provider_mangers): rename dataset-uris as asset-uris

Lee-W committed:
feat(providers/amazon): rename dataset-uris as asset-uris

Lee-W committed:
feat(providers/common/io): rename dataset-uris as asset-uris

Lee-W committed:
feat(providers/google): rename dataset-uris as asset-uris

Lee-W committed:
feat(providers/mysql): rename dataset-uris as asset-uris

Lee-W committed:
feat(providers/postgres): rename dataset-uris as asset-uris

Lee-W committed:
feat(providers/trino): rename dataset-uris as asset-uris

Lee-W committed:
test(dags): rename test_datasets.py as test_assets.py

Lee-W committed:
feat(listener/spec): rename airflow.listeners.spec.dataset as airflow.listeners.spec.asset

Lee-W committed:
feat(airflow/serialization/pydantic): rename airflow.serialization.pydantic.dataset as airflow.serialization.pydantic.asset

Lee-W committed:
feat(providers/mysql): rename airflow.providers.mysql.datasets as airflow.providers.mysql.assets

Lee-W committed:
feat(providers/postgres): rename airflow.providers.postgres.datasets as airflow.providers.postgres.assets

Lee-W committed:
feat(providers/trino): rename airflow.providers.trino.datasets as airflow.providers.trino.assets

Lee-W committed:
docs: update dataset as asset in documentation

Lee-W committed:
feat(jobs/scheduler_job_runner): rename dataset.orphaned as asset.orphaned

Lee-W committed:
feat(jobs/scheduler_job): rename dataset.triggered_dagruns as asset.triggered_dagruns

Lee-W committed:
feat(context): rename triggering_dataset_events as triggering_asset_events

Lee-W committed:
test(breeze): add common.compat to fab provider tests

Lee-W committed:
feat(providers/fab): add common.compt to fab dependencies due to RESOURCE_ASSET

Lee-W committed:
feat(providers): add back dataset-uris for backward compatibility

Lee-W committed:
feat(api_ui): rename views.datasets as views.assets

Lee-W committed:
docs(newsfragments): add type info

Lee-W committed:
build: update auto gen files

Lee-W committed:
feat(provider/microsoft): rename Dataset as Asset

Lee-W committed:
test(assets): fix asset hash error

Lee-W committed:
tests(providers): fix asset extra default to dict

mobuchowski commented on airflow/providers/openlineage/extractors/manager.py:
```diff
@@ -178,7 +178,7 @@ def extract_inlets_and_outlets(
     def get_hook_lineage(self) -> tuple[list[Dataset], list[Dataset]] | None:
         try:
-            from airflow.lineage.hook import get_hook_lineage_collector
+            from airflow.providers.common.compat.lineage.hook import get_hook_lineage_collector
```
You need to bump provider dependency for `common.compat` provider for it to work.

Lee-W replied:
Yep, the `common.compat` will be released with this PR, which is one of the reasons we can not bump it as it's not yet released. Had a discussion with TP. We'll not use compact in this PR but will bump it after the release.

jscheffl commented on tests/providers/amazon/aws/assets/test_s3.py:
```diff
@@ -20,13 +20,13 @@
-from airflow.providers.amazon.aws.datasets.s3 import (
-    convert_dataset_to_openlineage,
-    create_dataset,
+from airflow.providers.amazon.aws.assets.s3 import (
```
Such change would break the usage of the provider in Airflow prior 3.0.
Can you please make the changes in a way that tests are also able to run with Airflow 2.8-2.10?

That is for all `tests/providers/**`

jscheffl replied:
Besides this we also decided to have provider PRs always spearate from Core changes allowing to cherry-pick if needed.

Lee-W replied:
>  Can you please make the changes in a way that tests are also able to run with Airflow 2.8-2.10?

I think this should work fine for 2.8-2.10. We're running the tests against those versions as well. I fixed most of them

> Besides this we also decided to have provider PRs always spearate from Core changes allowing to cherry-pick if needed.

I'm not sure we're able to do that for this one. If we make core changes without changing the provider, it breaks the tests. We need to change the core, and the providers (with backward compact) all alone. Some of the backward compact will be added to common.compat providers. Once released, we'll remove the duplicate code and use the logic in common.compat provider

Lee-W committed:
build(fast_api): regen airflow/api_fastapi/openapi/v1-generated.yaml

Lee-W committed:
feat(providers/amazon): remove common.compat.asset dependency, we'll need to wait for the next common provider release

Lee-W committed:
feat(providers/fab): remove common.compat.asset dependency, we'll need to wait for the next common provider release

Lee-W committed:
feat(providers/openlineage): remove common.compat.asset dependency, we'll need to wait for the next common provider release

Lee-W committed:
test: remove common.compat deps from core tests

Lee-W committed:
feat(providers/common/io): remove common.compat.asset dependency, we'll need to wait for the next common provider release

Lee-W committed:
refactor(providers/common/compat): refactor asset compat get_hook_lineage_collector

Lee-W committed:
feat(providers/amazon): remove common.compat.lineage.asset_compat_lineage_collector dependency, we'll need to wait for the next common provider release

Lee-W committed:
feat(providers/openlineage): remove common.compat.lineage.asset_compat_lineage_collector dependency, we'll need to wait for the next common provider release

Lee-W committed:
style: fix mypy warnings

Lee-W committed:
feat(dag_processing): rename DatasetModelOperation as AssetModelOperation

Lee-W committed:
feat(assets): rename iter_dataset_aliases as iter_asset_aliases

Lee-W committed:
feat(simple_auth_manager): rename is_authorized_dataset as is_authorized_asset

uranusjr commented on .github/workflows/basic-tests.yml:
Rebase error?

Lee-W replied:
Opps... yep, this is something missed. Just removed it

uranusjr commented on RELEASE_NOTES.rst:
Same

Lee-W replied:
This is needed as there won't be dataset.html in the futre

uranusjr commented on airflow/api_fastapi/serializers/assets.py:
And this? (Not exactly sure, but this file does not exist in main now)

uranusjr commented on airflow/api_connexion/endpoints/dataset_endpoint.py:
Missed renaming this file

Lee-W replied:
This is intentionally kept this way and will be addressed in the ` (TODO) API endpoint change` step in https://github.com/apache/airflow/issues/42307

Lee-W committed:
build(ui): regen openapi files

uranusjr commented on airflow/config_templates/config.yml:
```diff
@@ -477,22 +477,22 @@ core:
-      description: Class to use as dataset manager.
-      version_added: 2.4.0
+    asset_manager_class:
+      description: Class to use as asset manager.
```
Should we mention this is renamed? (Is there a `version_changed` for this?)

Lee-W replied:
This is mentioned in `newsfragments/41348.significant.rst`. Is there any where you want us to mention it again?

Lee-W replied:
We just decide this would be a bad idea if we add it to description

uranusjr commented on airflow/lineage/__init__.py:
```diff
@@ -104,7 +104,7 @@ def prepare_lineage(func: T) -> T:
       if A -> B -> C and B does not have outlets but A does, these are provided as inlets.
     * "list of task_ids" -> picks up outlets from the upstream task_ids
-    * "list of datasets" -> manually defined list of data
+    * "list of assets" -> manually defined list of asset
```
I believe this refers to [OpenLineage datasets](https://openlineage.io/docs/spec/object-model/), not Airflow’s, so we should keep it as-is.

uranusjr replied:
(This is one of the many reasons why we want to rename, fwiw.)

Lee-W replied:
ah, got it. I misunderstood this one. 🤦‍♂️ Let me update it.

Lee-W committed:
test: fix rebase error in api endpoints

Lee-W committed:
fix(api_fastapi): remove unused assets.py

jscheffl commented on airflow/providers/amazon/aws/assets/s3.py:
Understood from previous comment response that it is required to change both providers and core here. But this very much smells like a breaking change in provider. Is is acceptable for the Airflow 2.10 community use?

Lee-W replied:
Yep, it would be a breaking change in provider, but it should still work for 2.10. If `airflow.assets` if not found (2.10), it'll fallback to dataset and use it as asset

jscheffl replied:
So if the provider itself is not a "public API" (is it?) and is is just an "internal breaking change" then... I would assume then it is not a breaking change, correct?

@eladkal how do we treat such changes, do we need to take care for external dependencies "in the wild" for the provider itself?

uranusjr replied:
We shouldn’t break packages that depend on providers if it’s viable. But in the case I don’t see why external packages would break?

jscheffl commented on airflow/providers/amazon/aws/auth_manager/avp/entities.py:
+1 breaking change here - Are we sure we can make the auth manager working with Airflow 2.10 with the `DATASET` constant removed?

Lee-W replied:
Once the user updates to this specific version, the internal code will use asset instead of dataset so I think it should work as it is.

jscheffl commented on airflow/providers/amazon/aws/auth_manager/aws_auth_manager.py:
```diff
@@ -161,15 +158,12 @@ def is_authorized_dag(
             context=context,
         )
 
-    def is_authorized_dataset(
```
+1 breaking change?

Lee-W replied:
Yes, this is a breaking change

jscheffl commented on docs/apache-airflow/img/asset-scheduled-dags.png:
Does the screenshot not need to have an update? The tag is still named "dataset" :-D

Lee-W replied:
This is intentionally kept this way. All the UI components and related docs will be updated in another PR.

jscheffl commented on docs/apache-airflow/img/assets.png:
+1 Screenshot refresh needed?

Lee-W replied:
This is intentionally kept this way. All the UI components and related docs will be updated in another PR.

jscheffl reviewed (commented):
Uff. Oh boy, this is a big one!

I fear there are some breaking changes in AWS provider. Maybe @uranusjr can you have an opinion on this?

Has the UI been tested if all is working? Some changes in UI very probably are not automatically be tested. Did not make a test and opened the UI. Might be a good point to double check not to break any link/HTML.

I have a shy approve, in general OK except the comments. For sure some more eyes are needed.

jscheffl reviewed (approved):
I approve from my side... but for sure as this is so large I'd request another pair of eyes for review before merge.

Lee-W committed:
feat(asset/manager): rename iter_dataset_alias, notify_dataset_alias_created as iter_asset_alias and notify_asset_alias_created

uranusjr reviewed (approved):
Admittedly I did not read every line, but _I think_ I read enough to be confident this is good…
